src/praisonai-train/praisonai_train/cli/commands/serve.py

```
@@ -0,0 +1,183 @@
+                port=port,
+                ngl=ngl,
+            )
+            typer.echo(f"llama-server running (pid {proc.pid}). Press Ctrl-C to stop.")
+            try:
+                proc.wait()
+            except KeyboardInterrupt:
+                proc.terminate()
+                typer.echo("\nStopped.")
+    except (ValueError, RuntimeError) as exc:
+        output.print_error(str(exc))
+        raise typer.Exit(1)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)